service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java

```diff
@@ -1,26 +1,48 @@
 package uk.nhs.adaptors.gp2gp.ehr.utils;
 
+import lombok.SneakyThrows;
+
+import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeFormatterBuilder;
-import java.util.Date;
+
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.InstantType;
 
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
 public class DateFormatUtil {
-    private static final String UK_ZONE_ID = "Europe/London";
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
-        .appendPattern("yyyyMMddHHmmss")
-        .toFormatter();
 
-    public static String formatDate(Date date) {
-        if (date == null) {
+    @SneakyThrows
+    public static String formatDateTimeType(DateTimeType dateTimeType) {
+        if (dateTimeType == null) {
             throw new EhrMapperException("Could not format date");
         }
 
-        return DATE_TIME_FORMATTER.format(
-            date.toInstant()
-                .atZone(ZoneId.of(UK_ZONE_ID))
-                .toLocalDateTime());
+        switch (dateTimeType.getPrecision()) {
+            case YEAR:
+                return localDateTime(dateTimeType, "yyyy");
+            case MONTH:
+                return localDateTime(dateTimeType, "yyyyMM");
+            case DAY:
+                return localDateTime(dateTimeType, "yyyyMMdd");
+            default:
+                return localDateTime(dateTimeType, "yyyyMMddHHmmss");
+        }
+    }
+
+    private static String localDateTime(DateTimeType effectiveDateTimeType, String pattern) {
+        return LocalDateTime.ofInstant(effectiveDateTimeType.toCalendar().toInstant(), ZoneId.of("Europe/London"))
```

Comment:
In my PR I created a date time utils method for instants. Maybe you could copy this change and use it here?